crates/chia-datalayer/src/merkle.rs

```diff
@@ -0,0 +1,1997 @@
+#[cfg(feature = "py-bindings")]
+use pyo3::{
+    buffer::PyBuffer,
+    exceptions::{PyAttributeError, PyValueError},
+    pyclass, pymethods, PyResult, Python,
+};
+
+use clvmr::sha2::Sha256;
+use num_traits::ToBytes;
+use std::cmp::Ordering;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::iter::zip;
+use std::mem::size_of;
+use std::ops::Range;
+
+type TreeIndex = u32;
+type Parent = Option<TreeIndex>;
```

Comment:
This doesn't seem like a useful type alias to me

Reply:
The intent is to explicitly remind folks that a parent is not known to be 'something' and that we are _not_ using a magic value like a `TreeIndex` of `-1` or something to represent the no-parent case.

I would expect this to be caught up in a general switch to the newtype pattern so even if I convinced you of the value, let's leave this discussion open.